app/models/topic.rb

```diff
@@ -585,6 +585,13 @@ def self.for_digest(user, since, opts = nil)
           SiteSetting.digest_suppress_categories.split("|").map(&:to_i),
         )
     end
+    if SiteSetting.digest_suppress_tags.present?
+      topics =
+        topics.joins("JOIN topic_tags tg ON topics.id = tg.topic_id").where(
```

Comment:
Is the join here going to inadvertently exclude untagged topics? Can you add a test case for that, just to be sure? Something like: 

- fabricate tagA
- create topic1 with tagA
- created topic2 without any tags
- set SiteSetting.digest_suppress_tags to include tagA
- expect Topic.for_digest to include only topic2